system_tests_topology: `AttributeError: 'TopologyFailover' object has no attribute 'set_state'` when reporting timeout from test

https://github.com/skupperproject/skupper-router/runs/6124438564?check_suite_focus=true#step:26:1609

```
    def timeout(self, name):
        if name == 'test':
>           self.set_state('Timeout Expired', 'bailing')
E           AttributeError: 'TopologyFailover' object has no attribute 'set_state'

/home/runner/work/skupper-router/skupper-router/skupper-router/tests/system_tests_topology.py:450: AttributeError
------------------------------ Captured log setup ------------------------------
INFO     proton:_reactor.py:1053 Connecting to Url('amqp://0.0.0.0:27354')...
INFO     proton:_reactor.py:1091 Connected to 0.0.0.0
INFO     proton:_reactor.py:1053 Connecting to Url('amqp://0.0.0.0:27355')...
INFO     proton:_reactor.py:1091 Connected to 0.0.0.0
INFO     proton:_reactor.py:1053 Connecting to Url('amqp://0.0.0.0:27356')...
INFO     proton:_reactor.py:1091 Connected to 0.0.0.0
INFO     proton:_reactor.py:1053 Connecting to Url('amqp://0.0.0.0:27353')...
INFO     proton:_reactor.py:1091 Connected to 0.0.0.0
INFO     proton:_reactor.py:1053 Connecting to Url('amqp://0.0.0.0:27353')...
INFO     proton:_reactor.py:1091 Connected to 0.0.0.0
INFO     proton:_reactor.py:1105 Disconnected, already closed
INFO     proton:_reactor.py:1053 Connecting to Url('amqp://0.0.0.0:27353')...
INFO     proton:_reactor.py:1091 Connected to 0.0.0.0
INFO     proton:_reactor.py:1105 Disconnected, already closed
INFO     proton:_reactor.py:1053 Connecting to Url('amqp://0.0.0.0:27353')...
INFO     proton:_reactor.py:1091 Connected to 0.0.0.0
INFO     proton:_reactor.py:1105 Disconnected, already closed
INFO     proton:_reactor.py:1053 Connecting to Url('amqp://0.0.0.0:27354')...
INFO     proton:_reactor.py:1091 Connected to 0.0.0.0
INFO     proton:_reactor.py:1105 Disconnected, already closed
INFO     proton:_reactor.py:1053 Connecting to Url('amqp://0.0.0.0:27354')...
INFO     proton:_reactor.py:1091 Connected to 0.0.0.0
INFO     proton:_reactor.py:1105 Disconnected, already closed
INFO     proton:_reactor.py:1053 Connecting to Url('amqp://0.0.0.0:27354')...
INFO     proton:_reactor.py:1091 Connected to 0.0.0.0
INFO     proton:_reactor.py:1105 Disconnected, already closed
INFO     proton:_reactor.py:1053 Connecting to Url('amqp://0.0.0.0:27355')...
INFO     proton:_reactor.py:1091 Connected to 0.0.0.0
INFO     proton:_reactor.py:1105 Disconnected, already closed
INFO     proton:_reactor.py:1053 Connecting to Url('amqp://0.0.0.0:27355')...
INFO     proton:_reactor.py:1091 Connected to 0.0.0.0
INFO     proton:_reactor.py:1105 Disconnected, already closed
INFO     proton:_reactor.py:1053 Connecting to Url('amqp://0.0.0.0:27355')...
INFO     proton:_reactor.py:1091 Connected to 0.0.0.0
INFO     proton:_reactor.py:1105 Disconnected, already closed
INFO     proton:_reactor.py:1053 Connecting to Url('amqp://0.0.0.0:27356')...
INFO     proton:_reactor.py:1091 Connected to 0.0.0.0
INFO     proton:_reactor.py:1105 Disconnected, already closed
INFO     proton:_reactor.py:1053 Connecting to Url('amqp://0.0.0.0:27356')...
INFO     proton:_reactor.py:1091 Connected to 0.0.0.0
INFO     proton:_reactor.py:1105 Disconnected, already closed
INFO     proton:_reactor.py:1053 Connecting to Url('amqp://0.0.0.0:27356')...
INFO     proton:_reactor.py:1091 Connected to 0.0.0.0
INFO     proton:_reactor.py:1105 Disconnected, already closed
------------------------------ Captured log call -------------------------------
INFO     proton:_reactor.py:1053 Connecting to Url('amqp://0.0.0.0:27353')...
INFO     proton:_reactor.py:1053 Connecting to Url('amqp://0.0.0.0:27354')...
INFO     proton:_reactor.py:1053 Connecting to Url('amqp://0.0.0.0:27354')...
INFO     proton:_reactor.py:1053 Connecting to Url('amqp://0.0.0.0:27355')...
INFO     proton:_reactor.py:1053 Connecting to Url('amqp://0.0.0.0:27356')...
INFO     proton:_reactor.py:1091 Connected to 0.0.0.0
INFO     proton:_reactor.py:1091 Connected to 0.0.0.0
INFO     proton:_reactor.py:1091 Connected to 0.0.0.0
INFO     proton:_reactor.py:1091 Connected to 0.0.0.0
INFO     proton:_reactor.py:1091 Connected to 0.0.0.0
---------------------------- Captured log teardown -----------------------------
INFO     proton:_reactor.py:1105 Disconnected, already closed
INFO     proton:_reactor.py:1105 Disconnected, already closed
INFO     proton:_reactor.py:1105 Disconnected, already closed
INFO     proton:_reactor.py:1105 Disconnected, already closed
- generated xml file: /__w/skupper-router/skupper-router/skupper-router/build/tests/junitxmls/system_tests_topology.xml -
=========================== short test summary info ============================
FAILED ::TopologyTests::test_01_topology_failover - AttributeError: 'Topology...
```

--- a/tests/system_tests_topology.py
+++ b/tests/system_tests_topology.py
@@ -447,7 +447,7 @@ class TopologyFailover (MessagingHandler):
     # ------------------------------------------------------------------------
     def timeout(self, name):
         if name == 'test':
-            self.set_state('Timeout Expired', 'bailing')
+            self.state_transition('Timeout Expired', 'bailing')
             self.bail("Timeout Expired: n_sent=%d n_received=%d n_accepted=%d" %
                       (self.n_sent, self.n_received, self.n_accepted))
         elif name == 'sender':

